extensions/cpu_adam/cpu_adam_x86.py

```diff
@@ -1,38 +1,27 @@
-from ..base_extension import BaseExtension
-from ..extension_builder import ExtensionBuilder
-from ..utils import append_nvcc_threads
-
-
-class X86CPUAdamExtension(BaseExtension):
-    def __init__(self) -> None:
-        super().__init__()
-        self.kernel_builder = X86CPUAdamBuilder()
-        self._requires_build = False
-
-    @property
-    def requires_build(self) -> bool:
-        return self._requires_build
+import platform
 
-    def build(self):
-        self.kernel_builder.build()
-        self._requires_build = True
+from ..cuda_extension import _CudaExtension
+from ..utils import append_nvcc_threads
 
-    def load(self):
-        return self.kernel_builder.load()
 
+class CpuAdamX86Extension(_CudaExtension):
```

Comment:
Why does cpu adam inherit cuda extention?

Reply:
Cuz CPU adam needs to include cuda header files.

<img width="464" alt="Screenshot 2024-01-24 at 15 17 18" src="https://github.com/hpcaitech/ColossalAI/assets/31818963/3763c65a-636a-4694-ad87-0ad4fcafc7d3">